### DISCUSSION TITLE
Updating Docs

### DISCUSSION THREAD

**Discussion by adnan29979** [Ylianst/MeshCentral#5926] [General]
An update of MeshCentral docs including setting up the SMTP server of Google (Gmail). (Which is currently taken care of by @buckybytes). 
And possibly other stuff as well.

**Comment by silversword411** [comment]
?
Do a PR, and update it.
https://github.com/Ylianst/MeshCentral/pulls

**Reply by si458** [reply]
Thats the idea.
This is a discussion thread for people who want to take on the task of updating docs.
:)

**Reply by buckybytes** [reply]
Well, it turns out I had to use my (Workspace/G-Suite/whateveritistoday) account to get it up and going on my live system.  

I've got all of the screenshots and text compiled for that process.

I'm going to get a dev environment up and running so I can use a throwaway Gmail to run the process with that.  Hopefully most of my screenshots and text will apply under those conditions too.  There's a couple of screenshots I'd like to replace, just in case I'm not as good at redacting as I think I am.

I see the first docs PR is queued.  Yay!

**Reply by adnan29979** [reply]
Ahh, I thought you were busy and **wouldn't be able to work with the Google SMTP documentation at all**. If you had commented a day later, I would have jumped on the Google SMTP documentation myself. Thanks for letting me know.

**Reply by buckybytes** [reply]
My instructions still work after 7 days, so I've converted the PR to ready.

**Comment by adnan29979** [comment]
The following picture is the view of '[https://my.domain.com:6969/translator.htm](https://meshcentral.da2i2.link:8069/translator.htm)'
How do I translate to a language which is not on this list using the `translator.htm` webpage? @si458 can you guide me please?

![image](https://github.com/Ylianst/MeshCentral/assets/148310766/543706c8-4879-4b49-855a-18752eec4cfb)

**Reply by si458** [reply]
this has been asked many a times, so we do indeed need this adding to the docs if you wouldnt mind plz, but here goes...

1. delete `translate.json` from inside of your `meshcentral-data` folder
(this is YOUR local copy of translations, but as we add new features/text, they will be missing from here)
2. login to meshcentral first!
3. visit `https://MYMESHCENTRALSERVER.COM/translator.htm`
4. select what language you would like to translate (your pic above)
5. use the search box OR scroll to find some text you would like to translate (or the `show no translations only` checkbox)
6. select some text, then enter the language translation in bottom box (not the top one), then click `SET (F1)`
7. repeat step 5+6 as many times as you like
8. once ready, click `SAVE TO SERVER (F3)` (this will save YOU a copy to `meshcentral-data/translate.json`)
9. (optional `SAVE TO FILE (F4)` will download the `translate.json` for you to look at or email or upload or whatever you want)
10. click `TRANSLATE SERVER` and sit back and have a brew! (takes about 5-15mins depending on your MESHSERVER specs)
11. once finished, RESTART MESHCENTRAL! (important for it to pick up the translated files correctly)

also note, IF you make ANY modifications to `default.handlebars` you need to run the `translate server` to copy the modifications over to the languages handlebar files (theses are stored in `node_modules/meshcentral/views/translations`)

**Reply by adnan29979** [reply]
Okay. Gotchu. Will add this to the docs. No worries 👌. You summarized it pretty well already. Thanks. 

i have a question tho. what if somebody wants to translate it into some language which is NOT in the language options to choose from (my pic above). 

edit: for example, i want to add a language translation which is not in the given list.

**Reply by si458** [reply]
erm... good question? not a bloody clue!? will have to check with @Ylianst on that one!

edit:
seems to be hardcoded in the `translator.htm`
https://github.com/Ylianst/MeshCentral/blob/31ebb21e0be8c31b424db42d7902fdddd64f549e/public/translator.htm#L62-L89

**Reply by adnan29979** [reply]
haha 😂! okay, sure. 
Please do let me know if u get any idea from Ylianst. I wish to add Bengali (now) and probably some other languages (in the future).

**Reply by si458** [reply]
do you know what the `shortcode` is for the languages?
i can try adding them for you!

**Reply by adnan29979** [reply]
'bn' for Bengali. 

edit: https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes

**Reply by si458** [reply]
beat me too it! was just about to ask is that list about right!

**Reply by adnan29979** [reply]
hehe. 
yea, seems about right. 

take a look here as well: https://www.loc.gov/standards/iso639-2/php/code_list.php

i think `ISO 639-2 Code` would be better to move forward with in the future since lot more languages can be represented using `ISO 639-2 Code` code. but for now, `ISO 639-1 Code` as it was should be just fine imo.

**Reply by si458** [reply]
that might be a BIG job, so ill start small.
try adding extra languages 2 code base `ISO 639-1 Code`,
then once all working, try for 3 code base instead.

**Reply by adnan29979** [reply]
Aight, gotchu. 
Thanks for working on it. looking forward to it.

**Reply by adnan29979** [reply]
@si458 added the documentation for translating meshcentral and added 'Bengali' to the list of hardcoded options in `translator.htm`. that seemed to have worked in my testing environment.

**Comment by adnan29979** [comment]
> seems to be hardcoded in the `translator.htm`

yup. saw this just now. seemed like hardcoded to me as well.
